api/serializers.py

```diff
@@ -96,7 +96,7 @@ class PuzzleSerializer(serializers.ModelSerializer):
     tags = serializers.SerializerMethodField(required=False)
     guesses = serializers.SerializerMethodField()
     # Have to specify this explicitly for validate_url to run
-    url = serializers.CharField()
+    url = serializers.URLField()
```

Comment:
Does this allow an empty url? I seem to recall we discussed at some point we might want to create a puzzle placeholder where the url was left blank (and perhaps filled in later).